airflow/operators/python_operator.py

```diff
@@ -167,21 +167,28 @@ def execute(self, context):
         logging.info('Skipping downstream tasks...')
         session = settings.Session()
 
+        downstream_tasks = context['task'].get_flat_relatives(upstream=False)
+        downstream_task_ids = [d.task_id for d in downstream_tasks]
+
+        logging.debug("Downstream task_ids {}".format(downstream_task_ids))
+
         TI = TaskInstance
         tis = session.query(TI).filter(
             TI.execution_date == context['ti'].execution_date,
-            TI.task_id.in_(context['task'].downstream_task_ids),
+            TI.task_id.in_(downstream_task_ids),
```

Comment:
Yes, you should also check `TI.dag_id == dag_id`, otherwise you can pull in other task instances with the same name but different dags, and the lookup may be a bit slower